modules/hal_sensors/i2c_sensor_driver/src/I2C_sensor_driver.hpp

```diff
@@ -24,51 +24,30 @@
  * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
  * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
  * OTHER DEALINGS IN THE SOFTWARE.
-***********************************************************************************************/
+ ***********************************************************************************************/
 
-#ifndef I2C_HELPER_HPP_
-#define I2C_HELPER_HPP_
-#include <stdint.h>
+#ifndef I2C_SENSOR_DRIVER_HPP
+#define I2C_SENSOR_DRIVER_HPP
 
-#ifdef __arm__
-#include <Wire.h>
-#define I2C_PERIPHERAL_T TwoWire*
-#else
-#include <i2c_peripheral_mock.hpp>
-#define I2C_PERIPHERAL_T I2CPeripheralMock*
-#endif
+#include <I2C_Driver.hpp>
 
-typedef enum {
-  kI2cSpeed_100KHz = 100000, kI2cSpeed_400KHz = 400000,
-} I2CSpeed;
-
-class I2CDriver {
+class I2C_sensor_driver : public hal::i2c::I2C_Driver {
```

Comment:
I hate this idea! It doesn't make the whole i2c easier to use.. Another class the compiler can't optimize away... even more memory usage. Because every sensor now has a private instance of this driver. That means double the memory usage of the previous solution. If you really want the read reg/write reg functions gone from i2c driver.. consider moving them as separate functions in to each source file of the sensors.. This is just a waste of resources.. Never gonna approve this change.. If you want to read why this is bad idea.. Read Philosophy of Software design page 46 and 62 and to some extent 25. This class duplicates functions or provides decorators which are in my eyes red flags